llvm/lib/MC/ELFObjectWriter.cpp

```diff
@@ -1088,13 +1090,14 @@ uint64_t ELFWriter::writeObject(MCAssembler &Asm, const MCAsmLayout &Layout) {
 
     // EVM local begin
     const MCSymbolELF *SignatureSymbol = Section.getGroup();
-    if (Section.getName() != ".text")
+    if (Ctx.getTargetTriple().isEVM() && Section.getName() != ".text")
```

Comment:
```suggestion
    if (Ctx.getTargetTriple().isEVM()) {
      if (Section.getName() == ".text")
        writeSectionData(Asm, Section, Layout);
      continue;
    }
```
If we do this, we can reduce number of places where we are adding local changes, since we can remove changes on line 1097. Wdyt?

Reply:
Thank you!